### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by revonateB0T**

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailScreen.kt [hunk 1]
```diff
@@ -592,22 +593,22 @@ fun GalleryDetailScreen(args: GalleryDetailScreenArgs, navigator: DestinationsNa
...
-            suspend fun showTorrentDialog() {
-                if (mTorrentList == null) {
-                    mTorrentList = bgWork {
-                        withIOContext {
-                            EhEngine.getTorrentList(galleryDetail.torrentUrl!!, gid, token)
-                        }
+            val torrentList = remember {
+                async(Dispatchers.IO) {
+                    runSuspendCatching {
+                        EhEngine.getTorrentList(galleryDetail.torrentUrl!!, gid, token)
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailScreen.kt:599]: Not sure about this.
Now users who don't use torrents will pay for what they don't need.

Comment (revonateB0T) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailScreen.kt:599]: Make it start lazily?

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailScreen.kt:599]: SGTM

File: app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailScreen.kt [hunk 2]
```diff
@@ -592,22 +594,22 @@ fun GalleryDetailScreen(args: GalleryDetailScreenArgs, navigator: DestinationsNa
...
                     }
                 }
-                if (mTorrentList!!.isEmpty()) {
+            }
+            suspend fun showTorrentDialog() {
+                val list = bgWork { torrentList.await() }.getOrThrow()
+                if (list.isEmpty()) {
                     showSnackbar(noTorrents)
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailScreen.kt:607]: nit: May change this to something like "No torrents for the current version (of the gallery)".

**Comment by revonateB0T** [comment]
Abandoned

**Pull request closed without merging**